mirage.opam

```diff
@@ -29,7 +29,6 @@ depends: [
   "alcotest" {with-test}
   "fmt" {>= "0.8.9" & with-test}
 ]
-available: opam-version >= "2.1.0"
```

Comment:
Note that it is required on `opam-repository` for two reasons:
1) we only able to use `opam.2.1`
2) the release on `opam-repository` uses the `hidden-flags` which is available only on `opam.2.1`

The real issue is mostly about `ocaml-ci` which fails to solve packages.

Reply:
This unblocks ocaml-ci so there is an issue here. I am not sure to understand why this is needed, I'll comment on the main PR.

Reply:
I agree that what we release to opam-repository should include the available constraint, but for this repository - if ocaml-ci is not yet on opam 2.1 - I see more value in allowing the CI to run here than to be pedantic about the availability.

would be great to also open an issue with ocaml-ci to update opam to 2.1.